### DISCUSSION TITLE
Add the updateType "unstable" / or some other form of "matchUnstableUpdate" as packageRule

### DISCUSSION THREAD

**Discussion by SeWieland** [renovatebot/renovate#26971] [Suggest an Idea]
I want to be able to configure something like

```json
{
  "packageRules": [
    {
      "matchUpdateTypes": ["unstable"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

I know this is probably complicated to do under the updateType, because there might be major, minor or patch pre-releases.
So probably another switch would work as well, e.g.

```json
{
  "packageRules": [
    {
      "matchUnstableUpdate": true,
      "dependencyDashboardApproval": true
    }
  ]
}
```

Plus a corresponding preset: `:approveUnstableUpdates`.

This functionality is currently missing completely and is especially painful, when automerge is active.

Right now, I'm limited to the following experience:

* Either use `ignoreUnstable: "true"` and do not receive pre-release updates at all
* Or use `ignoreUnstable: "false"` and receive PRs directly for unstable updates without any approval.

We had some cases for dependencies were we use pre-releases often and thus `ignoreUnstable: "false"`  is configured. We also configured `:automergePatch`. But then there were unstable _patch_ updates that we actually did not want to get merged...

**Comment by rarkins** [comment]
I think the root cause here is that any version change which is not major or minor is classified as "patch".

If we are using non-semver like 1.2.3.0 -> 1.2.3.1 then that can be referred to as a "build" update type (not yet supported.

If the update is from 1.2.3-beta.1 to 1.2.3-beta.2 then we should classify it as a "prerelease" updateType. If so then you can use matchUpdateTypes=prerelease. Would that solve your problem?

**Reply by SeWieland** [reply]
yes, that would absolutely work very well. The updateType "prerelease" is exactly what I need.
Also "prerelease" is probably more specific than "unstable" :+1:

**Reply by rarkins** [reply]
Converted to feature request issue: https://github.com/renovatebot/renovate/issues/26988
